Feature/drello board

eddybanks opened the pull request:
- Created the board page
- Added column and card components
- Added new column components
- Refactored global grid styling and default css margin and padding

eddybanks committed:
Add template images for boards

eddybanks committed:
Add custom links-item component for links and boards link

eddybanks committed:
Set up board page with basic board components

- Added components for columns, board navbar and column cards
- Removed unnecessary link-item component

eddybanks committed:
Merge branch 'development' into feature/drello-board

eddybanks committed:
Remove grid template  12 column structure

eddybanks committed:
Add links to dynamically generate boards page for each board item

eddybanks committed:
Fix board type to reflect that columns are optional as an attribute

eddybanks committed:
Get rid of default element margins and add form for new columns

eddybanks committed:
Adjust styling and refactor board card

eddybanks committed:
Add spacing to board navbar

noahdevx committed:
Add react-hook-form

eddybanks committed:
Add new board column components

- Moved the component responsible for new columns out of the board column component
- Tinkered a bit with styling

eddybanks committed:
Merge remote-tracking branch 'origin/features/add-react-hook-form' into feature/drello-board

eddybanks committed:
Correct onsubmit event type

eddybanks committed:
Merge branch 'bug/app-document-types' into feature/drello-board

noahdevx commented on src/pages/workspaces/index.tsx:
```diff
@@ -47,7 +48,14 @@ const Workspaces = () => {
-                <li key={drelloBoardsItem.id}>{drelloBoardsItem.title}</li>
+                <Link
+                  key={drelloBoardsItem.id}
+                  href={`/boards/${drelloBoardsItem.id}`}
```
Want to use constant instead

eddybanks replied:
I made the change. Thanks for the reminder!

noahdevx commented on src/pages/workspaces/index.tsx:
```diff
@@ -59,7 +67,14 @@ const Workspaces = () => {
-                <li key={focusBoardItem.id}>{focusBoardItem.title}</li>
+                <Link
+                  key={focusBoardItem.id}
+                  href={`/boards/${focusBoardItem.id}`}
```
Want to use constant instead

eddybanks replied:
I made the change. Thanks for the reminder!

nao-noah commented on src/pages/boards/[id].tsx:
```diff
@@ -0,0 +1,54 @@
+
+const Board = () => {
+  return (
+    <>
```
```suggestion
```

nao-noah commented on src/pages/boards/[id].tsx:
```diff
@@ -0,0 +1,54 @@
+          <NewBoardColumn />
+        </BoardContainer>
+      </BoardMain>
+    </>
```
```suggestion

```

noahdevx commented on src/components/workspaces/focus.tsx:
```diff
@@ -28,7 +30,11 @@ export const Focus = ({ drelloBoardsList }: FocusProps) => {
         {drelloBoardsList.map((workspaceItem, idx) => (
           <FcBdSummary key={idx}>
-            <BoardCard title={workspaceItem.title} />
+            <Link href={`/boards/${workspaceItem.id}`}>
```
We can use constant instead

noahdevx commented on src/components/workspaces/boards-list.tsx:
```diff
@@ -47,7 +50,11 @@ export const BoardsList = ({ drelloBoardsList }: BoardsListProps) => {
       <BdList>
         {drelloBoardsList.map((boardItem) => (
-          <BoardCard title={boardItem.title} key={boardItem.id} />
+          <Link key={boardItem.id} href={`/boards/${boardItem.id}`}>
```
We can use constants instead

noahdevx commented on src/components/boards/new-board-column.tsx:
```diff
@@ -0,0 +1,84 @@
+  );
+};
+
+const NewColumnContainer = styled.div`
```
We can set styled-components in the same order as otheres

eddybanks replied:
Sounds good!!

noahdevx commented on src/components/boards/board-navbar.tsx:
```diff
@@ -0,0 +1,28 @@
+export const BoardNavbar = () => {
+  return (
+    <NavMain>
+      <Link href="/">
```
We can use constants instead

noahdevx reviewed (commented):
Looks good overall but comment a little bit

eddybanks committed:
Resolve suggestions from pull request

- Used constants for url paths
- Created constant for landing path
- Moved styled components in new-board-column file.

eddybanks committed:
Merge branch 'development' into feature/drello-board
